[Task-2] Implement EF

Aleksander-Drumev-WA opened the pull request:
Domain models, DbContext and configurations created.

Aleksander-Drumev-WA committed:
Domain models' configurations

var1ap-WA commented on WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs:
```diff
@@ -0,0 +1,30 @@
+    {
+        public void Configure(EntityTypeBuilder<Basket> entity)
+        {
+            // Maybe unnecessary because has PK next line.
```
you are right

var1ap-WA replied:
remove this config for Id

Aleksander-Drumev-WA replied:
Done.

var1ap-WA commented on WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs:
```diff
@@ -0,0 +1,30 @@
+                .Property(b => b.Id)
+                .IsRequired();
+
+            entity.HasKey(b => b.Id);
```
we need to write base Entity class

var1ap-WA commented on WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs:
```diff
@@ -0,0 +1,30 @@
+
+            entity.HasKey(b => b.Id);
+
+            entity.HasOne(b => b.User)
```
you have forgotten about FK

Aleksander-Drumev-WA replied:
Now it's there.

var1ap-WA commented on WA.Pizza.Infrastructure/Data/Configurations/BasketEntityConfiguration.cs:
```diff
@@ -0,0 +1,30 @@
+            entity.HasOne(b => b.User)
+                .WithOne(b => b.Basket);
+
+            entity.Property(b => b.UserId)
+                .IsRequired(false);
```
We agreed that it's not required for basket

Aleksander-Drumev-WA replied:
Removed.

var1ap-WA commented on WA.Pizza.Infrastructure/Data/Configurations/OrderEntityConfiguration.cs:
```diff
@@ -0,0 +1,28 @@
+
+            entity
+                .Property(o => o.CreatedOn)
+                .HasColumnType("DATETIME2")
```
no need

Aleksander-Drumev-WA replied:
Removed.

var1ap-WA commented on WA.Pizza.Infrastructure/Data/Configurations/CatalogItemEntityConfiguration.cs:
```diff
@@ -0,0 +1,39 @@
+
+            entity
+                .Property(ci => ci.Name)
+                .HasColumnType("VARCHAR")
```
no need

Aleksander-Drumev-WA replied:
Removed.

Aleksander-Drumev-WA committed:
Fixed bugs mentioned in pull request

Aleksander-Drumev-WA committed:
Core project created.
Domain models moved into Core project.
Some namespaces changed.
Data services created and their abstraction.

Aleksander-Drumev-WA committed:
Interfaces for data services deleted.

Aleksander-Drumev-WA committed:
Order data service created.
Rename data services.

var1ap-WA commented on WA.Pizza.Core/ConstantValues.cs:
```diff
@@ -0,0 +1,7 @@
+{
+    public static class ConstantValues
+    {
+        public const int CATALOG_ITEM_NAME_MAX_LENGTH = 150;
```
remove it

var1ap-WA commented on WA.Pizza.Core/Models/Order.cs:
```diff
@@ -1,17 +1,16 @@
+namespace WA.Pizza.Core.Models
 {
-    public class Order
+    public class Order : BaseEntity
```
You also need `OrderItem` entity

Aleksander-Drumev-WA committed:
Added OrderItems to Order entity.
